pkg/instancegroups/instancegroups.go

```diff
@@ -477,14 +478,32 @@ func (c *RollingUpdateCluster) validateClusterWithTimeout(validateCount int) err
 		// Reset the success count; we want N consecutive successful validations
 		successCount = 0
 
-		// Wait before retrying
-		// TODO: Should we check if we have enough time left before the deadline?
-		time.Sleep(c.ValidateTickDuration)
+		// Wait before retrying in some cases
+		// TODO: Should we check if we have enough time left before the deadline? 
+		if shouldWaitBeforeRetry(result.Failures, group) {
+			time.Sleep(c.ValidateTickDuration)
+		}
```

Comment:
There's a logic error here: when shouldWaitBeforeRetry fails, instead of letting the validation succeed and proceeding with updating the next node it instead goes into a fast spinloop.

When all failures are inapplicable it should go through the code block that increments `successCount`.

Reply:
One question, even if the failures are inapplicable and we skip the wait period. We definitely want to log the failures right? I think we would need to change the log messages here if I am right. If the validation fails and the failures are inapplicable, showing the failure log messages and a cluster validated messages would be very unintuitive for the sys admins. 
I am planning to copy this block of code and add in the else stmt if shouldWaitBeforeRetry fails. I will change up the log messages to indicate that there **were** failures but they were ignored because they are in a different instance group.
```
			successCount++
			if successCount >= validateCount {
				klog.Info("Cluster validated.")
				return nil
			}
			klog.Infof("Cluster validated; revalidating in %s to make sure it does not flap.", c.ValidateSuccessDuration)
			time.Sleep(c.ValidateSuccessDuration)
			continue
```